dcc-portal-ui/app/scripts/compounds/js/compounds.js

```diff
@@ -360,6 +360,9 @@ angular.module('icgc.compounds.services', ['icgc.genes.models'])
           .then(function(compound) {
             _compoundEntity = _compoundEntityFactory(compound.plain());
             defer.resolve(_self);
+          }, function(){
+            Notify.hide();
+            $state.go('404', {page: 'compound', id: compoundId});
```

Comment:
Shouldn't this only happen on 404? What if it was a 500, would this still transition?